src/Tasks/Common/NETSdkMessage.cs

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Build.Framework;
+using System;
+
+namespace Microsoft.NET.Build.Tasks
+{
+    /// <summary>
+    /// Provides a localizable mechanism for logging messages with different levels of importance from the SDK targets.
+    /// </summary>
+    public class NETSdkMessage : MessageBase
+    {
+        public string Importance { get; set; } = "Normal";
+
+        protected override void LogMessage(string message)
+        {
+            MessageImportance importance =
+                (MessageImportance)Enum.Parse(typeof(MessageImportance), Importance, true);
```

Comment:
This is notoriously slow. It may not matter when we have one message per build (and only in previews at that) at first, but could get pretty brutal if we ever need to log a lot of volume. 

I suggest extracting to ParseImportance and just doing:

```
if (string.Equals(importance, "High", StringComparison.OrdinalIngoreCase)) 
   return MessageImportance.High;

if (string.Equals(importance, "Low", StringComparison.OrdinalIngoreCase))
  return MessageImportance.Low;

if (string.Equals(importance, "Normal", StringComparison.OrdinalIngoreCase))
  return MessageImportance.Normal;

throw new ArgumentOutRangeException(nameof(importance));
```

I'm not concerned about not supporting a (theoretical) future importance value, because we control all of the inputs and test would fail the first time you try to use it in new code, at which point you just update the parse method.

Reply:
Another option: default to high importance since that's all we pass and only bother adding when we need another value at which time we can think about perf. Simplifies code.

Reply:
Hmm, so built-in Message does it as you have it: https://github.com/Microsoft/msbuild/blob/master/src/Tasks/Message.cs#L138

And I guess having different behavior than Message without Importance arg wouldn't be good.

I'll leave it to your judgment. We can always improve perf if it becomes a problem.